pkg/cache/tas_cache_test.go

```diff
@@ -365,8 +370,51 @@ func TestFindTopologyAssignment(t *testing.T) {
 					},
 				},
 			},
+			leastAllocated: true,
 		},
-		"host required; single Pod fits in the host": {
+		"block required; 4 pods fit into one host each; MostAllocated": {
```

Comment:
Do you have a test case demonstrating situation when the choice affect "mid" level in 3 levels of hierarch? In other words we choose "MostAllocated" rack, and then the choice translates to choosing nodes under this rack.

Reply:
Do you mean something like that? Firstly we choose the block with the mostAllocated, then rack and host 

https://github.com/kubernetes-sigs/kueue/pull/4228/files#diff-85a93b2977c9efa55ca4846fdd219e22733e2b736c4fa3dba06f70739c1bbdfbR601-R627

Reply:
Yeah, and do you have a test case when the new strategy matters for the top level?

Reply:
Is this what you had in mind?
https://github.com/kubernetes-sigs/kueue/pull/4228/files#diff-85a93b2977c9efa55ca4846fdd219e22733e2b736c4fa3dba06f70739c1bbdfbR536-R586